internal/server.go

```diff
@@ -299,9 +314,47 @@ func (serv *Server) handleRequest(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	reqId := strings.TrimLeft(r.URL.Path, "/")
+	token := strings.TrimLeft(r.URL.Path, "/")
+	var reqId string
+	var secretKey [32]byte
+	if serv.encrypt {
+		tokenBytes, err := base58.Decode(token)
+		if err != nil {
+			log.WithFields(log.Fields{
+				"token": token,
```

Comment:
Same as in line 334; this might leak sensitive information.

Reply:
done